hyperapp.d.ts

```diff
@@ -0,0 +1,149 @@
+declare module "hyperapp" {
+  export interface VNodeData {
+    [key: string]: any
+  }
+
+  export interface VNode {
+    tag: string
+    data: VNodeData
+    children: string | VNode | Array<VNode>
+  }
+
+  export type viewFunction<State, Actions> = (
+    state: State,
+    actions: Actions
+  ) => VNode
+  export type emitFunction<Data> = (event: string, data?: Data) => Data
+  export type vnodeFunction = (
+    props: VNodeData,
+    children: string | VNode | Array<VNode>
+  ) => VNode
+
+  export interface Actions<State> {
+    [key: string]:
+      | ((
+          state?: State,
+          actions?: Actions<State>,
+          data?: any,
+          emit?: emitFunction<any>
+        ) => Partial<State> | any)
```

Comment:
in TypeScript `void` works a bit differently than other languages (C, Java etc); it's not just "no return value", but it's more like an alias for `undefined`. For example:
```ts
function a(): void {}
let b = a(); // strange, but this is allowed, and it's b === undefined
```
So I think it's ok to have `void` and left out `undefined` from our signature. I would also omit `any` because it makes the signature useless for type checking.